flake.nix

```diff
@@ -13,56 +13,79 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable-small";
-    flake-utils.url = "github:numtide/flake-utils";
-
-    flake-compat = {
-      url = "github:edolstra/flake-compat";
-      flake = false;
-    };
-
-    haskell-flake-utils = {
-      url = "github:ivanovs-4/haskell-flake-utils";
-      inputs.flake-utils.follows = "flake-utils";
-    };
   };
 
-outputs = { self, nixpkgs, flake-utils, haskell-flake-utils, flake-compat, ... }:
-  haskell-flake-utils.lib.simpleCabalProject2flake {
-    inherit self nixpkgs;
+  outputs = inputs:
+    with builtins;
+    let
+      lib = inputs.nixpkgs.lib;
+      pnames = [
+        "automaton"
+        "rhine"
+        "rhine-bayes"
+        "rhine-examples"
+        "rhine-gloss"
+        "rhine-terminal"
+      ];
+      supportedGhcs = [ "ghc92" "ghc94" "ghc96" "ghc98" ];
+      hpsFor = pkgs:
+        lib.filterAttrs (name: _: elem name supportedGhcs) pkgs.haskell.packages
```

Comment:
I think I don't really understand this. Isn't the aim to build this package for all supported GHCs?

Reply:
`hpsFor` converts `pkgs` into an attribute set of `{ ${ghcName} = pkgs.haskell.packages.${ghcName}; }` for every supported `ghcName`.

This allows us to easily map over this attribute set e.g. when creating `rhine-all` or per-GHC devshells.

Reply:
Makes sense! One detail: This silently drops those attribute names that are listed in `supportedGhcs` but do not occur in `pkgs.haskell.packages`? I think it would be better if there was a loud error when e.g. a ghc version is misspelled, or not maintained in nixpkgs anymore. Let me try and adapt that.

Reply:
For that I'd suggest replacing `filterAttrs` with `genAttrs`.